sensor/kubernetes/eventpipeline/resolver/resolver_impl.go

```diff
@@ -50,20 +49,35 @@ func (r *resolverImpl) runResolver() {
 // processMessage resolves the dependencies and forwards the message to the outputQueue
 func (r *resolverImpl) processMessage(msg *component.ResourceEvent) {
 	if msg.DeploymentReference != nil {
-		referenceIds := msg.DeploymentReference(r.deploymentStore)
+		referenceIds := msg.DeploymentReference(r.storeProvider.Deployments())
 
 		for _, id := range referenceIds {
-			preBuiltDeployment := r.deploymentStore.Get(id)
+			preBuiltDeployment := r.storeProvider.Deployments().Get(id)
 			if preBuiltDeployment == nil {
 				log.Warnf("Deployment with id %s not found", id)
 				continue
 			}
 
+			if msg.ParentResourceAction == central.ResourceAction_REMOVE_RESOURCE {
+				r.storeProvider.EndpointManager().OnDeploymentRemoveByID(id)
+				// At the moment we need to also send this deployment to the compatibility module when it's being deleted.
+				// Moving forward, there might be a different way to solve this, for example by changing the compatibility
+				// module to accept only deployment IDs rather than the entire deployment object. For more info on this
+				// check the PR comment here: https://github.com/stackrox/stackrox/pull/3695#discussion_r1030214615
+				event := component.NewResourceEvent([]*central.SensorEvent{toEvent(msg.ParentResourceAction, preBuiltDeployment)},
+					[]component.CompatibilityDetectionMessage{{Object: preBuiltDeployment, Action: msg.ParentResourceAction}}, nil)
```

Comment:
I completely overlooked that. I think we could survive by moving it back to the handler, but I should then also move the `endpointManager.OnDeploymentRemove` back since it relies on grabbing the `deploymentWrap` from the store. Alternatively we could remove deployments from the store in the resolver. I don’t really like either of these two solutions because we will not have a clear separation of responsibilities between components. I think the correct way to go is, as you mentioned, re-thinking the abstraction and maybe send a copy of the deployment to the resolver. Let’s sync about it.